Intermittent test failures in `next` branch

This test has been failing occasionally in `next` on the `mysql-python3.12` unittest step.

https://github.com/nautobot/nautobot/actions/runs/11806093120/job/32890279349

```
======================================================================
FAIL: test_list_objects_unknown_filter_no_strict_filtering (nautobot.extras.tests.test_views.ObjectMetadataTestCase.test_list_objects_unknown_filter_no_strict_filtering)
Verify that without STRICT_FILTERING, an unknown filter is ignored.
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/opt/hostedtoolcache/Python/3.12.7/x64/lib/python3.12/unittest/case.py", line 58, in testPartExecutor
    yield
  File "/opt/hostedtoolcache/Python/3.12.7/x64/lib/python3.12/unittest/case.py", line 634, in run
    self._callTestMethod(testMethod)
    ^^^^^^^^^^^^^^^^^
  File "/opt/hostedtoolcache/Python/3.12.7/x64/lib/python3.12/unittest/case.py", line 589, in _callTestMethod
    if method() is not None:
    ^^^^^^^^^^^^^^^^^
  File "/home/runner/.cache/pypoetry/virtualenvs/nautobot-zO_ZO7c3-py3.12/lib/python3.12/site-packages/django/test/utils.py", line 461, in inner
    return func(*args, **kwargs)
    ^^^^^^^^^^^^^^^^^
  File "/home/runner/work/nautobot/nautobot/nautobot/core/testing/views.py", line 799, in test_list_objects_unknown_filter_no_strict_filtering
    self.assertIn(instance2.get_absolute_url(), content, msg=content)
    ^^^^^^^^^^^^^^^^^
  File "/opt/hostedtoolcache/Python/3.12.7/x64/lib/python3.12/unittest/case.py", line 1152, in assertIn
    self.fail(self._formatMessage(msg, standardMsg))
    ^^^^^^^^^^^^^^^^^
  File "/opt/hostedtoolcache/Python/3.12.7/x64/lib/python3.12/unittest/case.py", line 715, in fail
    raise self.failureException(msg)
    ^^^^^^^^^^^^^^^^^
AssertionError: '/extras/object-metadata/69aa3c8c-8605-4cf1-8242-1a7652df21f4/' not found in  ...
```

--- a/changes/6509.fixed
+++ b/changes/6509.fixed
@@ -1,0 +1,2 @@
+Disallowed association of `ObjectMetadata` as metadata to other `ObjectMetadata` records.
+Removed unused object-detail view for `ObjectMetadata` records.

--- a/nautobot/core/testing/views.py
+++ b/nautobot/core/testing/views.py
@@ -1,3 +1,4 @@
+import contextlib
 import re
 from typing import Optional, Sequence
 from unittest import skipIf
@@ -792,11 +793,16 @@ class ViewTestCases:
             response = self.client.get(f"{self._get_url('list')}?id={instance1.pk}")
             self.assertHttpStatus(response, 200)
             content = utils.extract_page_body(response.content.decode(response.charset))
+            # There should be only one row in the table
+            self.assertIn('<tr class="even', content)
+            self.assertNotIn('<tr class="odd', content)
             if hasattr(self.model, "name"):
                 self.assertRegex(content, r">\s*" + re.escape(escape(instance1.name)) + r"\s*<", msg=content)
                 self.assertNotRegex(content, r">\s*" + re.escape(escape(instance2.name)) + r"\s*<", msg=content)
-            if instance1.get_absolute_url() in content:
-                self.assertNotIn(instance2.get_absolute_url(), content, msg=content)
+            with contextlib.suppress(AttributeError):
+                # Some models, such as ObjectMetadata, don't have a detail URL
+                if instance1.get_absolute_url() in content:
+                    self.assertNotIn(instance2.get_absolute_url(), content, msg=content)
 
         @override_settings(EXEMPT_VIEW_PERMISSIONS=["*"], STRICT_FILTERING=True)
         def test_list_objects_unknown_filter_strict_filtering(self):
@@ -833,11 +839,16 @@ class ViewTestCases:
             content = utils.extract_page_body(response.content.decode(response.charset))
             self.assertNotIn("Unknown filter field", content, msg=content)
             self.assertIn("None", content, msg=content)
+            # There should be at least two rows in the table
+            self.assertIn('<tr class="even', content)
+            self.assertIn('<tr class="odd', content)
             if hasattr(self.model, "name"):
                 self.assertRegex(content, r">\s*" + re.escape(escape(instance1.name)) + r"\s*<", msg=content)
                 self.assertRegex(content, r">\s*" + re.escape(escape(instance2.name)) + r"\s*<", msg=content)
-            if instance1.get_absolute_url() in content:
-                self.assertIn(instance2.get_absolute_url(), content, msg=content)
+            with contextlib.suppress(AttributeError):
+                # Some models, such as ObjectMetadata, don't have a detail URL
+                if instance1.get_absolute_url() in content:
+                    self.assertIn(instance2.get_absolute_url(), content, msg=content)
 
         @override_settings(EXEMPT_VIEW_PERMISSIONS=[])
         def test_list_objects_without_permission(self):

--- a/nautobot/extras/models/metadata.py
+++ b/nautobot/extras/models/metadata.py
@@ -201,6 +201,7 @@ class ObjectMetadata(ChangeLoggedModel, BaseModel):
     objects = ObjectMetadataManager()
     natural_key_field_names = ["pk"]
     documentation_static_path = "docs/user-guide/platform-functionality/objectmetadata.html"
+    is_metadata_associable_model = False
 
     class Meta:
         ordering = ["metadata_type"]

--- a/nautobot/extras/tables.py
+++ b/nautobot/extras/tables.py
@@ -978,6 +978,7 @@ class MetadataTypeTable(BaseTable):
 
 class ObjectMetadataTable(BaseTable):
     pk = ToggleColumn()
+    # NOTE: there is no identity column in this table; this is intentional as we have no detail view for ObjectMetadata
     metadata_type = tables.Column(linkify=True)
     assigned_object = tables.TemplateColumn(
         template_code=ASSIGNED_OBJECT, verbose_name="Assigned object", orderable=False

--- a/nautobot/extras/templates/extras/inc/job_table.html
+++ b/nautobot/extras/templates/extras/inc/job_table.html
@@ -28,7 +28,7 @@
                 </th>
             </tr>
             {% endifchanged %}
-            <tr class="collapseme-{{ row.record.grouping|slugify }}{% if not perms.extras.run_job or not row.record.runnable %} disabled{% endif %} collapse in" data-parent="#accordion" {{ row.attrs.as_html }}>
+            <tr class="{% cycle 'even' 'odd' %} collapseme-{{ row.record.grouping|slugify }}{% if not perms.extras.run_job or not row.record.runnable %} disabled{% endif %} collapse in" data-parent="#accordion" {{ row.attrs.as_html }}>
                 {% for column, cell in row.items %}
                     <td {{ column.attrs.td.as_html }}>{{ cell }}</td>
                 {% endfor %}

--- a/nautobot/extras/tests/test_views.py
+++ b/nautobot/extras/tests/test_views.py
@@ -3033,8 +3033,6 @@ class ObjectChangeTestCase(TestCase):
 
 
 class ObjectMetadataTestCase(
-    ViewTestCases.GetObjectViewTestCase,
-    ViewTestCases.GetObjectChangelogViewTestCase,
     ViewTestCases.ListObjectsViewTestCase,
 ):
     model = ObjectMetadata

--- a/nautobot/extras/views.py
+++ b/nautobot/extras/views.py
@@ -2228,8 +2228,6 @@ class MetadataTypeUIViewSet(NautobotUIViewSet):
 
 
 class ObjectMetadataUIViewSet(
-    ObjectChangeLogViewMixin,
-    ObjectDetailViewMixin,
     ObjectListViewMixin,
 ):
     filterset_class = filters.ObjectMetadataFilterSet

